Minor fixes for NVIDIA Ampere support

magnumripper opened the pull request:
Document (in code) Ampere's CC and try to recognize it even on macOS.
Change the wording in --list=opencl-devices to "CUDA INT32 cores" as there might be cores supporting only FP and so on.

Closes #4892

magnumripper committed:
Minor fixes for NVIDIA Ampere support

Document (in code) Ampere's CC and try to recognize it even on macOS.
Change the wording in --list=opencl-devices to "CUDA INT32 cores" as there
might be cores supporting only FP and so on.

Closes #4892

solardiz commented on src/opencl_common.c:
```diff
@@ -2541,8 +2541,8 @@ cl_uint get_processors_count(int sequential_id)
-		// Volta or Turing
-		else if (strstr(dname, "TITAN V") || strstr(dname, "RTX 2"))
+		// Volta, Turing, Ampere
+		else if (strstr(dname, "TITAN V") || strstr(dname, "RTX 2") || strstr(dname, "RTX 3"))
```
We should also detect "RTX A", and in fact I think it'd be safer to make it just "RTX " instead of listing the 2, 3, A prefixes. Yes it would also detect future RTX series, but I think it's probably best to guess the same as previous known revision.

We can also do `dname[0] == 'A' && dname[1] >= '1' && dname[1] <= '9'` for the A100, etc.

magnumripper committed:
Comment fix for OpenCL "speed index"

If CPU or "unknown" GPU, we factor the vector width for long.  A comment
had it as we used half that figure, but we don't - I was thinking half
the one for int - see #3502.
